pandas/core/ops.py

```diff
@@ -505,12 +509,18 @@ def _convert_to_array(self, values, name=None, other=None):
         inferred_type = lib.infer_dtype(values)
         if (inferred_type in ('datetime64', 'datetime', 'date', 'time') or
                 is_datetimetz(inferred_type)):
+
             # if we have a other of timedelta, but use pd.NaT here we
             # we are in the wrong path
             if (supplied_dtype is None and other is not None and
-                (other.dtype in ('timedelta64[ns]', 'datetime64[ns]')) and
+                    (other.dtype in ('timedelta64[ns]', 'datetime64[ns]')) and
                     isna(values).all()):
-                values = np.empty(values.shape, dtype='timedelta64[ns]')
+                if len(values) == 1 and other.dtype == 'timedelta64[ns]':
```

Comment:
this is not simpler at all.

```
values = np.empty(values.shape, dtye=other.dtype)
values[:] = iNaT
```

Reply:
I agree, hence the sarcasm quotes and the simpler original implementation.

Reply:
Yah, even then this is wrong because it isn't conditioning on the op being __sub__.  Gonna revert back to how it was before this commit.